phase-4-explainer.md

```diff
@@ -0,0 +1,175 @@
+
+```
+interface ComputedAccessibleNode {
+    readonly attribute DOMString? autocomplete;
+    readonly attribute DOMString? checked;
+    readonly attribute DOMString? keyShortcuts;
+    readonly attribute DOMString? name;
+    readonly attribute DOMString? placeholder;
+    readonly attribute DOMString? role;
+    readonly attribute DOMString? roleDescription;
+    readonly attribute DOMString? valueText;
+
+    readonly attribute long? colCount;
+    readonly attribute unsigned long? colIndex;
+    readonly attribute unsigned long? colSpan;
+    readonly attribute unsigned long? level;
+    readonly attribute unsigned long? posInSet;
+    readonly attribute long? rowCount;
+    readonly attribute unsigned long? rowIndex;
+    readonly attribute unsigned long? rowSpan;
+    readonly attribute long? setSize;
+
+    readonly attribute ComputedAccessibleNode? parent;
+    readonly attribute ComputedAccessibleNode? firstChild;
+    readonly attribute ComputedAccessibleNode? lastChild;
+    readonly attribute ComputedAccessibleNode? previousSibling;
+    readonly attribute ComputedAccessibleNode? nextSibling;
+
+    readonly attribute boolean? atomic;
+    readonly attribute boolean? busy;
+    readonly attribute boolean? disabled;
+    readonly attribute boolean? modal;
+    readonly attribute boolean? readOnly;
+
+  readonly attribute float? valueNow;
+  readonly attribute float? valueMin;
+  readonly attribute float? valueMax;
+
+    [CallWith=ScriptState] Promise ensureUpToDate();
+};
+```
+
+A full list of available attributes can be found [on this trix]() (aboxhall@: should this list be migrated somewhere it can be viewed externally? Putting it into the explainer seems wrong, should it be uploaded as a separate markdown or filed into an issue on the wicg page once the explainer is updated there?).
+
+
+##### 2. Rejection Case
+If we are returning a promise when requesting a `ComputedAccessibleNode`, what is the rejection case?
+
+##### 3. Lifecycle of the Accessible Tree
+`ComputedAccessibleNode`s within the same Frame all share a reference to the same tree snapshot, which means the tree is created when the first node is requested, and is destroyed with the Frame. One alternative is to queue up a number of requests for `ComputedAccessibleNode`s somewhere (returning a pending `Promise` to the callee), and when the next layout/style recalc is performed, resolve any outstanding `Promise`s. Computing the tree lazily would allow for batch retrieval of `ComputedAccessibleNode`s but care would have to be taken the ensure the idea of a consistent snapshot is maintained.
+
+##### 4. Accessibility Setting
+The Accessibility setting must be enabled for Phase 4. The current proposed implementation forces this when a `ComputedAccessibleNode` is requested, and does not ever switch it off, which may not be the best option in the long term. The lifecycle of Phase 4 from the first request of a `ComputedAccessibleNode`
```

Comment:
This is an implementation detail that doesn't need to be in this doc.

Reply:
Removed.